### PR TITLE
Refresh balances upon each tx confirmation

### PR DESCRIPTION
**Pull request by pakaplace**
ALC-180
ALC-165
* Replace tai chi documentation
* Add refreshBalances function (which also refreshes crucibles) to callback of `EVENT.txConfirmed`
* Prevent contract calls from being made if on wrong network by adding network checks
* Refresh upon network change

### CODE REVIEW

File: src/components/crucible/detail/rewards.tsx [hunk 1]
```diff
@@ -168,12 +167,12 @@ const Rewards: React.FC<Props> = ({ crucible }) => {
             <VStack justifyContent='center' pt={4}>
               <Button
                 width='100%'
-                disabled={!(crucible.cleanUnlockedBalance > 0)}
+                disabled={crucible.cleanUnlockedBalance <= 0}
```
Comment (nuurdev) [inline src/components/crucible/detail/rewards.tsx:170]:
```suggestion
                disabled={Number(crucible.cleanUnlockedBalance) <= 0}
```

Comment (nuurcodes) [inline src/components/crucible/detail/rewards.tsx:170]:
```suggestion
                disabled={Number(crucible.cleanUnlockedBalance) <= 0}
```

File: src/components/crucible/detail/rewards.tsx [hunk 2]
```diff
@@ -168,12 +167,12 @@ const Rewards: React.FC<Props> = ({ crucible }) => {
...
                 onClick={() => setWithdrawModalOpen(true)}
               >
                 <Text fontSize='md'>Withdraw LP balance</Text>
               </Button>
-              {!(crucible.cleanUnlockedBalance > 0) && (
+              {crucible.cleanUnlockedBalance <= 0 && (
```
Comment (nuurcodes) [inline src/components/crucible/detail/rewards.tsx:175]:
```suggestion
              {Number(crucible.cleanUnlockedBalance) <= 0 && (
```

File: src/context/crucibles/crucibles.tsx [hunk 3]
```diff
@@ -106,89 +116,97 @@ const CruciblesProvider = ({ children }: CruciblesProps) => {
...
-                  resolve(
-                    calculateMistRewards(
-                      signer,
-                      Number(reward.currStakeRewards)
-                    )
-                  );
+  useEffect(
+    () => {
+      if (tokenBalances && provider && network) {
+        // process.env.REACT_APP_NETWORK_ID
```
Comment (nuurcodes) [inline src/context/crucibles/crucibles.tsx:138]: Can we remove this comment?

File: src/hooks/useContract.tsx [hunk 4]
```diff
@@ -57,6 +59,8 @@ export function useContract(
             }}
           />
         );
+        monitorTx(event.txHash, reloadBalances);
```
Comment (nuurcodes) [inline src/hooks/useContract.tsx:62]: Will this reload balances twice?

**Review by nuurcodes — APPROVED** [review]
Left some minor comments

File: src/context/crucibles/crucibles.tsx [hunk 5]
```diff
@@ -106,89 +116,97 @@ const CruciblesProvider = ({ children }: CruciblesProps) => {
...
+
+        const signer = provider.getSigner();
+        setIsRewardsLoading(true);
+        getOwnedCrucibles(signer, provider)
+          .then((ownedCrucibles: Crucible[]) => {
```
Comment (CryptoOsanu) [inline src/context/crucibles/crucibles.tsx:143]: Do you think that this whole piece of code could be written with async / await? It seems to me that it is quite difficult to understand and maintain it ( maybe it is just me )

File: src/hooks/useContract.tsx (hunk 4 above)
Comment (pakaplace) [reply src/hooks/useContract.tsx:62]: It'll reload upon tx confirmation and upon tx success. I'll remove the "upon tx confirmation" reload.

File: src/context/crucibles/crucibles.tsx (hunk 5 above)
Comment (pakaplace) [reply src/context/crucibles/crucibles.tsx:143]: It's certainly not the best but async/await would be my asynchronous (more stopping/awaiting for functions that have no dependent results).

**Pull request merged**